Lib/test/test_dataclasses/__init__.py

```diff
@@ -1411,6 +1413,61 @@ class C:
         C().x
         self.assertEqual(factory.call_count, 2)
 
+    def test_default_factory_with_no_init_method(self):
```

Comment:
Don't you need a test with a user-defined `__init__`? Both with `init=True` and `init=False`.

Reply:
I tested the init=False case but forgot about the init=True case. I'll update the test name as well, thanks for spotting it!